sw/src/pistackmond.cpp

```diff
@@ -145,6 +145,9 @@ std::vector<long int> getIntsFromLine(std::string s) {
 float fetchTemp() {
 	// Returns CPU temperature in degrees C
 
+#if defined (ODROID_C1)
+  return 0.0;
```

Comment:
Not sure where Armbian keeps the temperature, but it must be somewhere!
Perhaps the way to proceed is to handle "file not found" error and try in Armbian's path in that case.
Or even better, test it once on launch, so it won't try to read nonexisting file a few times per second.
It is really surprising, because I tested pistackmond successfully on both Ubuntu and Manjaro with no problems.

Reply:
Also, about the "defined" part.. Weren't you working on using PLATFORM variable instead?

Reply:
the PLATFORM variable contains ODROID_C1.

Regarding the temperature: this is not Armbian-specific. This is probably a problem of the given kernel-build or device-tree or whatever. I will try another version. There is no other place for the temp-file. Armbian is very proud of using a mainline kernel where possible.

Due to the #if define, the code never tries to read a nonexisting file. I would also bet that the optimizer removes the fetchTemp() function as a whole. It actually is a quick workaround to test the basic functionality and should be removed with a better solution.

Reply:
I see, so this is just me not understanding how defines actually work. :) What a learning opportunity!

So perhaps the best way to proceed is to harden fetchTemp() against cases like that. How about make it return 0 if file not found, and write a warning in stderr? A future user complaining that temperature sensing doesn't work could be asked to run pistackmond in a terminal to see if there are any warnings.

Reply:
yes. Ideally, the daemon would write into the syslog as well.

Another thing I remembered that I skipped automatic hardware configuration during system-initialization. Maybe this is the reason for the missing temp-file. I will check this on Thursday.